history/storage.py

```diff
@@ -1,48 +1,69 @@
 from __future__ import unicode_literals
 
-from datetime import datetime
-import logging
 import boto
 import base64
+from datetime import datetime
 import json
+import logging
 import urllib
 
-from scrapy import log
 from scrapy.conf import settings
-from scrapy.utils.request import request_fingerprint
-from scrapy.responsetypes import responsetypes
+from scrapy.exceptions import NotConfigured
 from scrapy.http import TextResponse, Headers
+from scrapy.responsetypes import responsetypes
+from scrapy.utils.request import request_fingerprint
+
+MANDATORY_SETTINGS = ['HISTORY_S3_BUCKET',
+                      'AWS_ACCESS_KEY_ID',
+                      'AWS_SECRET_ACCESS_KEY']
 
 
 logger = logging.getLogger(__name__)
 
+
 class S3CacheStorage(object):
 
     def __init__(self, stats, settings=settings):
-        # Required settings
+        # Mandatory settings
         self.S3_ACCESS_KEY = settings.get('AWS_ACCESS_KEY_ID')
         self.S3_SECRET_KEY = settings.get('AWS_SECRET_ACCESS_KEY')
         self.S3_CACHE_BUCKET = settings.get('HISTORY_S3_BUCKET', None)
+        configured = all([settings.get(k, False) for k in MANDATORY_SETTINGS])
+        if not configured:
+            raise NotConfigured('% are mandatoy settings, set them either from the settings file '
+                                'or from the Scrapinghub spider settings '
+                                'section.' % ','.join(MANDATORY_SETTINGS))
 
         # Optional settings
-        self.use_proxy = settings.get('HISTORY_USE_PROXY', True)
+        # boto s3_connection does not work through proxy.
+        # comment this line from the original file
+        # self.use_proxy = settings.get('HISTORY_USE_PROXY', False)
         self.SAVE_SOURCE = settings.get('HISTORY_SAVE_SOURCE',
-                                        '{name}/{ts}__{jobid}')
+                                        '{name}/{time}__no_job_id')
```

Comment:
So we will use the same path for all jobs without `job_id`. What are the consequences ? is this default sane ?